src/daemon/plugin.c

```diff
@@ -583,12 +583,12 @@ static void *plugin_read_thread(void __attribute__((unused)) * args) {
      * should be called. */
     rf->rf_next_read += rf->rf_effective_interval;
 
-    /* Check, if `rf_next_read' is in the past. */
-    if (rf->rf_next_read < now) {
-      /* `rf_next_read' is in the past. Insert `now'
-       * so this value doesn't trail off into the
-       * past too much. */
-      rf->rf_next_read = now;
+    /* Check, if `rf_next_read' is on the same second as 'now' or in the past */
+    if (CDTIME_T_TO_TIME_T(rf->rf_next_read) <= CDTIME_T_TO_TIME_T(now)) {
```

Comment:
Why the addition of `CDTIME_T_TO_TIME_T()`?

Reply:
CDTIME_T_TO_TIME_T() to round timestamp to nearest integer second.
E.g.:
now = 1667983304.2
rf->rf_next_read = 1667983304.4
It means, that current metric is collected at 1667983304.2 and next one will be collected at 1667983304.4
Technically, condition `rf->rf_next_read < now` is false but if we don't increment rf->rf_next_read, server gets metrics with these timestamps in sequence: 1667983304.2, 1667983304.4. They both round to 1667983304, but rrdtool storage "resolution" is 1 second minimum and rrd rejects whole batch of data, causing gap on graph.

Reply:
Collectd supports intervals smaller than one second. With this change, we'd cut that off.